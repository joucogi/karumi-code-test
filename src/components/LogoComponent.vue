<template>
  <div>
    <img alt="Karumi logo" src="../assets/logo.png" />
  </div>
</template>
<style scoped>
  img {
    width: 60%;
    min-width: 80px;
  }
</style>