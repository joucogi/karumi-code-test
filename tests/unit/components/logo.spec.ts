import { shallowMount, Wrapper } from "@vue/test-utils";
import Vue from "vue";

import LogoComponent from "@/components/LogoComponent.vue";
import { toMatchSnapshot } from "../helpers";

let wrapper: Wrapper<Vue>;

beforeEach(() => {
  wrapper = shallowMount(LogoComponent);
});

describe("Logo Component", () => {
  it("renders image logo when passed", () => {
    expect(wrapper.html()).toContain(
      '<img alt="Karumi logo" src="../assets/logo.png">'
    );
  });

  it("renders correctly", () => {
    toMatchSnapshot(wrapper);
  });
});
