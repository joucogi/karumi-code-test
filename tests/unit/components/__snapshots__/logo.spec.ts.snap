// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Component renders correctly 1`] = `
<img
  alt="Karumi logo"
  src="../assets/logo.png"
/>
`;
